### PR TITLE
Add jq language support

### PR DESCRIPTION
**Pull request by valpackett**
jq is a language for manipulating JSON data: https://jqlang.github.io/jq/

Screenshot from 2024-08-01 00-13-28

(wild code from the screenshot: https://github.com/wader/jqjq)

### CODE REVIEW

File: runtime/queries/jq/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/jq/highlights.scm:1]: The highlights need some small changes to adjust to the captures we use: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

For example `property` should become `variable.other.member`, `boolean` should become `constant.builtin.boolean`, etc. Also the `spell` capture can be removed - neovim mixes in their spell checking queries into their highlights queries. We don't currently have spell checking queries but when introduced they'll be separate from highlights

File: runtime/queries/jq/locals.scm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+;; From nvim-treesitter, contributed by @ObserverOfTime et al.
+
+(funcdef
+  (identifier) @local.definition.function)
+
+(funcdefargs
+  (identifier) @local.definition.parameter)
```
Comment (the-mikedavis) [inline runtime/queries/jq/locals.scm:7]:
```suggestion
  (identifier) @local.definition)

(funcdefargs
  (identifier) @local.definition)
```

For locals we only support `local.scope`, `local.definition` and `local.reference` captures

File: runtime/queries/jq/highlights.scm
Comment (valpackett) [reply runtime/queries/jq/highlights.scm:1]: Right, I have changed a few of those, but missed several as well. Adjusted these too

Comment (the-mikedavis) [reply runtime/queries/jq/highlights.scm:1]: Did you mean to include that change in the most recent push? I don't mean to rush you to make the change - it sounds like you've already made the change but the most recent push might've accidentally only been a rebase

Comment (valpackett) [reply runtime/queries/jq/highlights.scm:1]: oops, lol, pushed the actual thing now

File: runtime/queries/jq/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+; Labels
+(query
+  label: (variable) @label)
+
+(query
+  break_statement: (variable) @label)
+
+; Literals
+(number) @number
```
Comment (the-mikedavis) [inline runtime/queries/jq/highlights.scm:24]:
```suggestion
(number) @constant.numeric
```

**Review by the-mikedavis — COMMENTED** [review]
Just one other capture needs an update, otherwise this looks good

**Review by the-mikedavis — APPROVED** [review]
Thanks!

**Comment by wader** [comment]
@valpackett just noticed this, thanks for adding jq support!

**Pull request merged**